.github/workflows/rust.yaml

```diff
@@ -21,7 +21,7 @@ jobs:
         with:
           components: clippy
       - name: Lint
-        run: cargo clippy -- -D warnings
+        run: cargo clippy --workspace --all-targets --tests -- -D warnings
```

Comment:
as documented in the [github section](https://doc.rust-lang.org/stable/clippy/continuous_integration/github_actions.html) of the clippy documentation, let's use
```
cargo clippy --all-targets --all-features -- -D warnings
```

Reply:
done.